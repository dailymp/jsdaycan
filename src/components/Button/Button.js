import React from 'react';
import styled, { withTheme } from 'styled-components';

import Box from '../Box';

/**
 * Styled Component
 */
const StyledButton = styled(Box)`
  &:hover {
    background-color: ${({ theme, bgHover }) => theme.colors[bgHover]};
    border-color: ${({ theme, borderColor }) => theme.colors[borderColor]};
  }
`;

const Button = ({
  variant, size, theme, ...props
}) => {
  const color = variant === 'white' ? 'blackLight' : 'white';
  const colorHover = variant === 'white' ? 'black' : 'white';
  const bg = `${variant}Light`;
  const bgHover = variant;
  const borderColor = variant === 'white' ? 'blackLight' : variant;
  const borderColorHover = variant === 'white' ? 'black' : `${variant}Dark`;
  return (
    <StyledButton
      color={color}
      colorHover={colorHover}
      bg={bg}
      bgHover={bgHover}
      borderColor={borderColor}
      borderColorHover={borderColorHover}
      {...theme.defaults.buttonSizes[size]}
      is="button"
      blacklist={[
        'colorHover',
        'bgHover',
        'borderColor',
        'borderColorHover',
      ]}
      {...props}
    />
  );
};

Button.defaultProps = {
  variant: 'mediumGray',
  size: 'md',
};

export default withTheme(Button);
