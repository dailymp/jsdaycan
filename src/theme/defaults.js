const defaults = {
  fontChangeBp: 1,
  buttonSizes: {
    sm: {
      py: [1, 2],
      px: [3, 4],
      fontSize: [1, 2],
    },
    md: {
      py: [2, 3],
      px: [4, 5],
      fontSize: [2, 3],
    },
    lg: {
      py: [3, 4],
      px: [6, 7],
      fontSize: [3, 4],
    },
  },
};

export default defaults;
