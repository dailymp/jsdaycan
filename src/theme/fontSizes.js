const fontSizes = [13, 14, 16, 18, 20, 24, 28];

export default fontSizes;
