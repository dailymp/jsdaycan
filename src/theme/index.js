import colors from './colors';
import fonts from './fonts';
import fontWeights from './fontWeights';
import sizes from './sizes';

const theme = {
  colors,
  fonts,
  fontWeights,
  sizes,
};

export default theme;
