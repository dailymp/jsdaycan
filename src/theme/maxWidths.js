const maxWidths = {
  main: '1200px',
};

export default maxWidths;
