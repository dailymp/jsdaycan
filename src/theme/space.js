const space = [0, 5, 10, 15, 20, 25, 35, 50, 100];

export default space;
