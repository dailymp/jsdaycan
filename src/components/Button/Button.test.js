import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Button from './Button';
import theme from '../../theme';

describe('Button', () => {
  const onClick = jest.fn();
  const variant = 'primary';
  const buttonComp = (
    <ThemeProvider onClick={onClick} theme={theme}>
      <Button variant={variant}>CHILD</Button>
    </ThemeProvider>
  );
  test('styles', () => {
    const tree = renderer.create(buttonComp).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('background-color', theme.colors.primaryLight);
    expect(tree).toHaveStyleRule('font-size', '16px');
    expect(tree).toHaveStyleRule('padding-left', '20px');
    // we can pass a modifier which will be applied before checking the assertion
    expect(tree).toHaveStyleRule('background-color', theme.colors.primary, {
      modifier: ':hover',
    });
    expect(tree).toHaveStyleRule('font-size', '18px', {
      media: 'screen and (min-width:40em)',
    });
    expect(tree).toHaveStyleRule('padding-left', '25px', {
      media: 'screen and (min-width:40em)',
    });
  });
});
