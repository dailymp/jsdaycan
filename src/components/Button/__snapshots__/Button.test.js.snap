// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button styles 1`] = `
.c1 {
  padding-left: 20px;
  padding-right: 20px;
  padding-top: 10px;
  padding-bottom: 10px;
  color: #fff;
  background-color: #45d2ae;
  font-size: 16px;
  border-color: #31CEA6;
}

.c0:hover {
  background-color: #31CEA6;
  border-color: #31CEA6;
}

@media screen and (min-width:40em) {
  .c1 {
    padding-left: 25px;
    padding-right: 25px;
    padding-top: 15px;
    padding-bottom: 15px;
  }
}

@media screen and (min-width:40em) {
  .c1 {
    font-size: 18px;
  }
}

<button
  bg="primaryLight"
  className="c0 c1"
  color="white"
  fontSize={
    Array [
      2,
      3,
    ]
  }
  px={
    Array [
      4,
      5,
    ]
  }
  py={
    Array [
      2,
      3,
    ]
  }
>
  CHILD
</button>
`;
